src/main/java/seedu/application/model/application/interview/InterviewComparator.java

```diff
@@ -9,16 +10,26 @@
  * fields. Applications are sorted by interview date first then interview time if date is the same.
  */
 public class InterviewComparator implements Comparator<Application> {
+
+    private static int DUMMY_INT = 0;
+
     @Override
-    public int compare(Application i1, Application i2) {
-        int compareDate = i1.getInterview().get().getInterviewDate().value
-                .compareTo(i2.getInterview().get().getInterviewDate().value);
-        int compareTime = i1.getInterview().get().getInterviewTime().value
-                .compareTo(i2.getInterview().get().getInterviewTime().value);
-        if (compareDate == 0) {
-            return compareTime;
+    public int compare(Application a1, Application a2) {
+        Optional<Interview> i1 = a1.getInterview();
+        Optional<Interview> i2 = a2.getInterview();
+        if (i1.isPresent() && i2.isPresent()) {
+            int compareDate = i1.get().getInterviewDate().value
+                    .compareTo(i2.get().getInterviewDate().value);
+            int compareTime = i1.get().getInterviewTime().value
+                    .compareTo(i2.get().getInterviewTime().value);
+            if (compareDate == 0) {
+                return compareTime;
+            } else {
+                return compareDate;
+            }
         } else {
-            return compareDate;
+            assert false : "Interview should exist while using comparator to sort";
```

Comment:
I guess for now it seems that this is not always true with the bug that @teekaytai spotted?

Reply:
Hmmm, I'm not sure how the code will change after he fix the bug but according to the current implementation the interview should exist. After checking whether the interview isPresent() in the Optional the UI exception in #126 will only not appear. 

Anyway if it doesn't exist it should also be our problem right because we should check the existence of interview before trying to sort it with the comparator? I'm putting this assertion by judging it would be our problem if using the interview comparator before checking its existence.

Reply:
oh yeah sorry I meant it as a comment but yup the logic makes sense here, there should be interviews to compare if we are using the comparator

Reply:
Oh nvm haha, actually I just checked the modified version of InterviewComparator and it actually no need this assert anymore as it handles the case where no interview is in the application.